<launch>

    <!--- Sim Time
    <param name="/use_sim_time" value="true" />
     -->

    <!--- Run Rviz-->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find lego_loam)/launch/test.rviz" />

    <!--- TF -->
    <node pkg="tf" type="static_transform_publisher" name="camera_init_to_map"  args="0 0 0 1.570795   0        1.570795 /map    /camera_init 10" />
    <node pkg="tf" type="static_transform_publisher" name="base_link_to_camera" args="0 0 0 -1.570795 -1.570795 0        /camera /base_link   10" />

    <!--- LeGO-LOAM
     <node pkg="lego_loam" type="pointPublish" name="pointPublish" output="screen" />
     -->
    <node pkg="lego_loam" type="imageProjection"    name="imageProjection"    output="screen"/>
    <node pkg="lego_loam" type="featureAssociation" name="featureAssociation" output="screen"/>
    <node pkg="lego_loam" type="mapOptmization"     name="mapOptmization"     output="screen"/>
    <node pkg="lego_loam" type="transformFusion"    name="transformFusion"    output="screen"/>


    <!--   是否使用GPS      -->
    <param name="useGPS" type="bool"   value="false"/>
    <!--   是否使用RANSAC          -->
    <param name="useRANSAC" type="bool"  value="false"/>

    <!--   KITTI序列号和数据地址-->
    <param name="seqNum" type="string" value="01"   />
    <param name="KITTI_directory" type="string" value="/media/localization/Localization/KITTIdataset" />

    <!--   发布KITTI数据-->
    <node pkg="lego_loam" type="pubKittiData" name="pubKittiData" output="screen"/>

     <!--   存储lego_loam里程计位姿信息到outputName.txt文件中  outputDirectory是存储目录   -->
    <node pkg="lego_loam" type="getOdometry" name="getOdometry" output="screen">
    </node>
    <arg name="outputDirectory" default="/media/localization/新加卷/school/allSchool"  />
    <param  name="outputName" value="$(arg outputDirectory).txt" />
<!--
 <node name="recorder" pkg="rosbag" type="record" args="/laser_cloud_surround /integrated_to_init -O $(arg outputDirectory)" />
-->

</launch>
